### PR TITLE
Add documentation to openshift/sdn

### PR DESCRIPTION
**Pull request by danielmellado**
This commit adds sphinx documentation, alongside a Dockerfile for
building it, to openshift/sdn repository.

It mainly has the current wiki as a source, and it aims to be blend
together with the code in order for it to be updated together.

### CODE REVIEW

File: docs/source/openshift_sdn/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+This is openshift-sdn, the default network plugin for OpenShift (both
+OKD and OCP). It uses Open vSwitch to connect pods locally, with VXLAN
+tunnels to connect different nodes.
+
+OpenShift SDN is designed to be installed by the [OpenShift Network
+Operator](https://github.com/openshift/cluster-network-operator), and
```
Comment (juanluisvaladas) [inline docs/source/openshift_sdn/README.md:6]: I think everything after the comma should be removed.

File: docs/source/openshift_sdn/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+certain components of it (such as the Deployment and DaemonSet
+objects) are found there.
+
+This module defines two images: the sdn image, which contains OpenShift SDN
+(both controller and node components) and the kube-proxy image, which is
+deployed by the Network Operator for third-party network plugins that need it.
+(Kube-proxy is built from here rather than from the origin repo so
```
Comment (juanluisvaladas) [inline docs/source/openshift_sdn/README.md:13]: If there previous sentence is finished with a "." I don't think we actually need the parenthesis

File: docs/source/openshift_sdn/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+Network Operator creates the CRD definitions.
+
+## The OpenShift SDN Controller
+
+The network controller is run on
+the masters to handle cluster-level processing:
+
+  - Creating `NetNamespace` objects corresponding to `Namespace`s
+  - Creating `HostSubnet` objects corresponding to `Node`s
+  - Implementing high availability for egress IPs
```
Comment (juanluisvaladas) [inline docs/source/openshift_sdn/README.md:37]: Don't you mean egressCIDRs ?

File: docs/source/openshift_sdn/workflow.rst [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+Just query cluster-bot and send him a message with the following::
+
+  launch openshift/sdn#<PR_NUMBER> aws
+
+Update your CNO image without deleting your cluster
+---------------------------------------------------
+
+.. code-block:: shell-session
+
+  update_cno ()
```
Comment (juanluisvaladas) [inline docs/source/openshift_sdn/workflow.rst:89]: you're not updating it, you're making it unmanaged, a better name would be unmanage_cno.

**Review by juanluisvaladas — CHANGES_REQUESTED** [review]
I'm not quite sure about this making sense here rather than the CNO. Or even if this should be part of CNO wiki...

**Comment by danwinship** [comment]
I guess I don't get how this is supposed to be an improvement over just having the markdown files? Why would I want to build a container and run a web server locally when I can just read the files, or view them off github.com?

File: docs/source/openshift_sdn/README.md (hunk 1 above)
Comment (danielmellado) [reply docs/source/openshift_sdn/README.md:6]: This content is just migrated, please don't go over it yet.

File: docs/source/openshift_sdn/README.md (hunk 2 above)
Comment (danielmellado) [reply docs/source/openshift_sdn/README.md:13]: ditto

File: docs/source/openshift_sdn/README.md (hunk 3 above)
Comment (danielmellado) [reply docs/source/openshift_sdn/README.md:37]: ditto

File: docs/source/openshift_sdn/workflow.rst (hunk 4 above)
Comment (danielmellado) [reply docs/source/openshift_sdn/workflow.rst:89]: ditto

**Comment by danielmellado** [comment]
> I guess I don't get how this is supposed to be an improvement over just having the markdown files? Why would I want to build a container and run a web server locally when I can just read the files, or view them off github.com?

Hi Dan, so overall let me give you some context. 

1) This content will be published to github pages once we agree on a proper location.

2) Sphinx [1], which is used by a lot of open source projects allows for a lot of functionalities, besides just having the files over github.

As a matter of fact, you can just go over the files themselves even if you don't want to run the container, which is just being provided as a commodity in order for a developer to be able to trigger such builld. Enhanced funcionality goes from search to code It also does provide a way to organize the content, which IMHO is lacking in just 'bare' markdown.

3) Having the documentation in-tree also allows us to do a proper branching and tracking of it per-release.

[1] https://www.sphinx-doc.org/en/master/

**Pull request merged**